meson.build

```diff
@@ -45,6 +45,63 @@ if get_option('enable_debugger') == 'heavy'
   conf_data.set10('C_HEAVY_DEBUG', true)
 endif
 
+# prepare configuration for dynrec core
+#
+# host_machine is a machine we're building for
+#
+# In code we have code paths for two additional architectures:
+#
+# - ARMV4LE - autoconf used to set it when "armv6l" was reported, presumably
+#   it covers everything up to ARMv7.  Notably, Raspberry Pi 1 is using ARMv6.
+#   This arch was marked as NOT handling unaligned memory.
+# - MIPSEL - which used to be called MIPSEL32, and seems to be never hooked
+#   into the autoconf buildsystem (but it was used by several DOSBox ports)
+#
+selected_core = get_option('dynamic_core')
+host_cpu = host_machine.cpu_family()
+conf_data.set('C_TARGETCPU', 'UNKNOWN')
+cpu_handles_unaligned_memory = false
+
+core_selection = [
+  # cpu         selected_core       dynrec_define     C_TARGETCPU
+  [ 'x86_64',  ['auto', 'dyn-x86'], 'C_DYNAMIC_X86',  'X86_64'    ],
+  [ 'x86',     ['auto', 'dyn-x86'], 'C_DYNAMIC_X86',  'X86'       ],
+  [ 'x86_64',  ['dynrec'],          'C_DYNREC',       'X86_64'    ],
+  [ 'x86',     ['dynrec'],          'C_DYNREC',       'X86'       ],
+  [ 'aarch64', ['auto', 'dynrec'],  'C_DYNREC',       'ARMV8LE'   ],
+  [ 'arm',     ['auto', 'dynrec'],  'C_DYNREC',       'ARMV7LE'   ],
+  [ 'ppc64',   ['auto', 'dynrec'],  'C_DYNREC',       'PPC64LE'   ], # stable name
+  [ 'ppc64le', ['auto', 'dynrec'],  'C_DYNREC',       'PPC64LE'   ], # unstable name
+  [ 'ppc',     ['auto', 'dynrec'],  'C_DYNREC',       'POWERPC'   ],
+]
+
+foreach line : core_selection 
+  cpu = line[0]
+  opts_for_arch = line[1]
+  dynrec_define = line[2]
+  target_cpu = line[3]
+
+  if (host_cpu == cpu) and opts_for_arch.contains(selected_core)
+    conf_data.set('C_TARGETCPU', target_cpu)
+    conf_data.set(dynrec_define, 1)
+    cpu_handles_unaligned_memory = true
+  endif
+endforeach
+
+conf_data.set10('C_UNALIGNED_MEMORY', cpu_handles_unaligned_memory)
```

Comment:
Line 63 has: `cpu_handles_unaligned_memory = false`
So this statement here always sets `false` for handling `C_UNALIGNED_MEMORY`?

Or maybe the unaligned memory handling ability per architecture will be added to the `core_selection` table?

Reply:
It sets `true` (which is converted to `1` in define due to usage of `set10` function).

All architectures we have defined ATM are marked as supporting unaligned memory in autoconf. We do build for s390x - I doubt anyone actually uses this arch for running DOSBox, but we are verifying codepaths for arch requiring aligned memory this way.

Old buildsystem marks these two options as *not* supporting unaligned memory:

```
   armv6l)
    AC_DEFINE(C_TARGETCPU,ARMV4LE)
    AC_MSG_RESULT(ARMv6 Little Endian)
    c_targetcpu="arm"
    dnl c_unalignedmemory=yes
    ;;
```
and
```
   *)
    AC_DEFINE(C_TARGETCPU,UNKNOWN)
    AC_MSG_RESULT(unknown)
    c_unalignedmemory=no
```

"armv6l" is weird and problematic ATM - the code was written for "ARMv4 Little Endian":

1) First problem is: meson does not differentiate between so old ARM versions - it reports 32-bit as `arm` (which is usually ARMv7) and 64-bit ARMv8 as `aarch64`. My guess is that we should probably move detection of older ARM variants to the code itself.
2) I do not understand the mess around current ARM code in cpu module - the code is written in several variants for "ARMv4", which is then supposed to be the same as ARMv6.

I couldn't find any references to ARMv4 - maybe it was some old version from 90s or something. ARMv6 is architecture from early 00s - still used in some embedded controllers and the like. The only device that we might want to support using this arch is Raspberry Pi 1 (but I do own RPi1 and tried to run dosbox on it once - it was starting up for 15 minutes and was totally unusable).

Reply:
I moved code working with CPU-related configure defines to `src/cpu/meson.build` and included unaligned memory in the table - hopefully it's more readable now.

Reply:
Thanks for this explanation. It's fantastic CI does support these more obscure architectures.

The reorganized code looks great too.